app/assets/javascripts/govuk_publishing_components/components/accordion.js

```diff
@@ -48,13 +48,18 @@ window.GOVUK.Modules.GovukAccordion = window.GOVUKFrontend.Accordion;
 
     // look for data attributes to put onto the 'show/hide all' link
     var showAllAttributes = this.$module.getAttribute('data-show-all-attributes')
+
     if (showAllAttributes) {
       try {
         var showAll = this.$module.querySelector('.' + this.showAllControls)
         var values = JSON.parse(showAllAttributes)
         var keys = Object.keys(values)
         for (var i = 0; i < keys.length; i++) {
-          showAll.setAttribute('data-' + keys[i], values[keys[i]])
+          if (typeof values[keys[i]] === 'object') {
+            showAll.setAttribute('data-' + keys[i], JSON.stringify(values[keys[i]]))
+          } else {
+            showAll.setAttribute('data-' + keys[i], values[keys[i]])
+          }
```

Comment:
As discussed, turns out this change isn't necessary - the existing code will already append any given attribute onto the show all element.